### PR TITLE
DB-9844 Spark 3.0 Remove org.spark_project.* references

### PR DESCRIPTION
**Pull request by ipraznik-splice**

### CODE REVIEW

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1060/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1064/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: remote_it/pom.xml [hunk 1]
```diff
@@ -77,10 +77,8 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <!-- org.spark_project.guava -->
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-network-common_${scala.binary.version}</artifactId>
-            <version>${spark.version}</version>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (hatyo) [inline remote_it/pom.xml:81]: shouldn't we also specify the version?

File: splice_access_api/pom.xml [hunk 2]
```diff
@@ -41,12 +41,9 @@
...
             <version>${project.version}</version>
         </dependency>
-
         <dependency>
-            <!-- org.spark_project.guava -->
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-network-common_${scala.binary.version}</artifactId>
-            <version>${spark.version}</version>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (hatyo) [inline splice_access_api/pom.xml:46]: ditto

File: splice_encoding/pom.xml [hunk 3]
```diff
@@ -42,15 +42,13 @@
...
-            <version>${spark.version}</version>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>annotations</artifactId>
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <groupId>com.google.code.findbugs</groupId>
-            <artifactId>annotations</artifactId>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (hatyo) [inline splice_encoding/pom.xml:51]: ditto

File: splice_machine/pom.xml [hunk 4]
```diff
@@ -86,10 +86,8 @@
             </exclusions>
         </dependency>
         <dependency>
-            <!-- org.spark_project.guava -->
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-network-common_${scala.binary.version}</artifactId>
-            <version>${spark.version}</version>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (hatyo) [inline splice_machine/pom.xml:90]: ditto

File: splice_si_api/pom.xml [hunk 5]
```diff
@@ -57,10 +57,13 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <!-- org.spark_project.guava -->
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-network-common_${scala.binary.version}</artifactId>
-            <version>${spark.version}</version>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <scope>provided</scope>
```
Comment (hatyo) [inline splice_si_api/pom.xml:62]: ditto

File: splice_timestamp_api/pom.xml [hunk 6]
```diff
@@ -34,10 +34,8 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <!-- org.spark_project.guava -->
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-network-common_${scala.binary.version}</artifactId>
-            <version>${spark.version}</version>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (hatyo) [inline splice_timestamp_api/pom.xml:38]: same

File: utilities/pom.xml [hunk 7]
```diff
@@ -40,15 +40,22 @@
...
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-network-common_${scala.binary.version}</artifactId>
-            <version>${spark.version}</version>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```
Comment (hatyo) [inline utilities/pom.xml:49]: same

File: remote_it/pom.xml (hunk 1 above)
Comment (ipraznik-splice) [reply remote_it/pom.xml:81]: default version is specified in the parent pom.xml:
 <guava.version>14.0.1</guava.version>
or can be changed in the profile

File: pom.xml [hunk 8]
```diff
@@ -92,6 +92,7 @@
         <netty-all.version>4.1.17.Final</netty-all.version>
         <spark.warehouse>-Dsplice.spark.sql.warehouse.dir=${project.build.directory}/spark-warehouse</spark.warehouse>
         <hbase.rootdir>-Dhbase.rootdir=${project.build.directory}/hbase</hbase.rootdir>
+        <guava.version>14.0.1</guava.version>
```
Comment (OlegMazurov) [inline pom.xml:95]: This version affects all profiles.
If guava is provided its version must match the one available at runtime.
For CDH 6.3.2, HBase is run with guava-11.0.2.jar while Spark loads two versions: guava-11.0.2.jar and guava-14.0.1.jar.
Two potential problems may occur: 
* Splice code running in HBase doesn't find a new method/field only available in 14.0.1
* deserialization in Spark fails to match a class due to version difference between HBase and Spark

**Comment by jyuanca** [comment]
This PR can be closed.

**Comment by ipraznik-splice** [comment]
Another implementation is prepared for review

**Pull request closed without merging**